### ISSUE TITLE
A question about widely misused finalizers and dispose

### ISSUE THREAD

**Issue by quixoticaxis** [MonoGame/MonoGame#6279]
I've been delving in the code and I've found a lot of classes that implement Dispose pattern not correctly.

1. Some of them try to emulate destructor semantics with a finalizer (which is considered an anti-pattern) to interact with global singletons. Although it is ugly and unreadable and a few classes do it in an unsafe way, (for example, `Game` with `ANDROID` defined, `GamePlatform`, etc.), I can understand their existence.
2. Some classes implement finalizes that do nothing useful at all. They are either empty or delegate their functions to some helper methods that do nothing meaningful for finalization (`VideoContent`, `SoundEffect`, `SoundEffectInstance`, `AudioEngine`, `SoundBank`, `WaveBank`, `ContentManager`, etc.) 
3. `GameComponent` defines completely meaningless finalizer that can only call the logic defined in his children, which is not needed because the children should provide their own finalizers if they work with unmanaged resources
4. Some classes seem to rely on the order of finalization (they try to dispose other entities through references and it is only meaningful if the entities have not been finalized) which is undetermined.

To cut the long story short, I've found few classes with finalizers that actually work with `GCHanldes` or unmanaged pointers.

Can someone explain me the reason why? Is it a natural artefact of the long going development (as I see in the history, some of the finalizers were removed/deprecated once, but came back later) or is their any greater meaning to them?

**Comment by quixoticaxis** [comment]
@Jjagg could you please provide a little guidance to me? As i started refactoring finalizers, I'm obviously in need of running tests. Is `MonoGameTests.exe` a confessional way of running tests? Also what result should I expect?
I've got
```
Test Run Summary
  Overall result: Warning
  Test Count: 1292, Passed: 1272, Failed: 0, Warnings: 0, Inconclusive: 0, Skipped: 20
    Skipped Tests - Ignored: 20, Explicit: 0, Other: 0
```
Are there any other tests I should run?

**Comment by Jjagg** [comment]
@quixoticaxis Yes, those are the tests you should run. If you set up a pull requests the build bot will run the tests as well, on Windows and on Mac, so you don't *have* to run them locally.

**Comment by quixoticaxis** [comment]
Created a PR #6656
